### ISSUE TITLE
False `missing-field` diagnostics in certain union types involving `enum`s

### ISSUE THREAD

**Issue by wcjohnson** [EmmyLuaLs/emmylua-analyzer-rust#1152]
## Environment
Windows 11, VSCode 1.127.0, Language Server Extension `tangzx.emmylua` version 0.9.39

## Observed Behavior
The following code:
```lua
---@enum NiceEnum
local GOODGUYS = {
	superman = 1
}

---@alias Evil string|NiceEnum

---@param evils Evil|(Evil[])
local function do_evil(evils) end

local function bad() do_evil({ "hi", "dead" }) end
```
will *sometimes* trigger the shown diagnostic:
<img width="1021" height="178" alt="Image" src="https://github.com/user-attachments/assets/84cc3b30-d0fd-4425-984c-3dc9f2518618" />

Whether the diagnostic is triggered appears to be nondeterministic. Restarting the emmylua language server sometimes makes it disappear and sometimes re-triggers it.

## Expected Behavior
That diagnostic should never trigger.

**Comment by wcjohnson** [comment]
For the record I don't believe it is simply a random LSP or cache error. Changing the types involved makes a difference. If `NiceEnum` is replaced with `int` for example, the error never triggers.

There is some underlying type checking issue with unions of enums, perhaps then exacerbated by caching problems as well.
